samples/volumes/example2.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import kfp.dsl as dsl
+
+
+@dsl.pipeline(
+    name="Example 2",
+    description="The second example of the design doc."
+)
+def example2():
+    vol_common = dsl.PipelineVolume(
+        name="mypvc",
+        size="10Gi",
+        mode=dsl.VOLUME_MODE_RWM
+    )
+
+    step1 = dsl.ContainerOp(
+        name="step1",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["echo 1 | tee /mnt/file1"],
+        volumes={"/mnt": vol_common}
+    )
+
+    step2 = dsl.ContainerOp(
+        name="step2",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["echo 2 | tee /mnt2/file2"],
+        volumes={"/mnt2": vol_common}
+    )
+
+    step3 = dsl.ContainerOp(
+        name="step3",
+        image="library/bash:4.4.23",
+        command=["sh", "-c"],
+        arguments=["cat /mnt/file1 /mnt/file2"],
+        volumes={"/mnt": vol_common.after(step1, step2)}
```

Comment:
I am not sure if this code is really needed. Users achieve same effect by: step3.after(step1, step2).

Reply:
+1

Reply:
@hongye-sun 
> I am not sure if this code is really needed. Users achieve same effect by: step3.after(step1, step2).

This is the only comment we don't really agree with.
Apart from being a bit unfortunately named -- I'll open a separate issue for this -- I understand `step3.after()` is meant to be a last resort; as a way to declare dependencies when there is no other way for the compiler to deduce them.

But what is great about using `PipelineParam` instances is that the compiler can *detect* dependencies based on how we use `PipelineParam` instances, i.e., when we format strings based on them.

Similarly, this PR aims to make volumes a first-class way of passing GBs of objects between steps, thus creating dependencies implicitly.

We argue that `PipelineParam` instances is the only way to deduce dependencies now, we will soon have full artifacts [and we are happy to contribute in their implementation!], similarly volumes should be such a way. So, yes, a `PipelineVolume` is a `V1Volume`, agreed, but it also brings in *dependencies* when used.

Let's talk more about this after I amend #801.